test/over_react/util/prop_conversion_test.dart

```diff
@@ -0,0 +1,1069 @@
+          expect(jsifyAndUnjsify(object), same(object));
+        });
+      });
+    });
+
+    group(
+        'works as expected when using converting prop getters/setters with JS components:',
+        () {
+      group('Map props using (un)jsifyMapProp', () {
+        group('get converted to JS objects', () {
+          group('in the setter, and gets unconverted in getter', () {
+            // This case is a little redundant with the (un)jsifyMapProp tests above, but include it for completeness.
+            test('when set to a Map', () {
+              final builder = TestJs()..buttonProps = {'data-foo': 'bar'};
+
+              final propKey = TestJs.getPropKey((p) => p.buttonProps);
+              expect(builder, {propKey: isA<JsMap>()},
+                  reason:
+                      'test setup: should have converted to a JS object for storage in props map'
+                      ' (we want to ensure this happens before it gets to the ReactComponentFactoryProxy)');
+              expect(builder.buttonProps, isA<Map>(),
+                  reason:
+                      'should have unconverted JsMap to a Map in the typed getter');
+            });
+
+            // This case is a little redundant with the (un)jsifyMapProp tests above, but include it for completeness.
+            test('when null', () {
+              final builder = TestJs();
+
+              expect(builder, {}, reason: 'test setup check');
+              expect(builder.buttonProps, isNull);
+
+              builder.buttonProps = null;
+              final propKey = TestJs.getPropKey((p) => p.buttonProps);
+              expect(builder, {propKey: null});
+              expect(builder.buttonProps, isNull);
+            });
+
+            group('and allows pattern of setting a map prop in a builder', () {
+              test('and then mutating the value read from the builder', () {
+                final builder = TestJs();
+
+                builder.buttonProps = {};
+                expect(builder.buttonProps, {});
+
+                final propKey = TestJs.getPropKey((p) => p.buttonProps);
+                expect(builder, {propKey: isA<JsMap>()},
+                    reason:
+                        'test setup: should have converted to a JS object for storage in props map');
+
+                // For this test, it's important to mutate the the value returned from the `builder.buttonProps` getter,
+                // and not the original Map passed into `builder.buttonProps`.
+                builder.buttonProps['data-foo'] = 'bar';
+                expect(builder.buttonProps, {'data-foo': 'bar'});
+              });
+
+              group('and then reading a nested value that gets unconverted:',
+                  () {
+                group('A nested map prop', () {
+                  test('using the typed props map', () {
+                    final builder = TestJs()
+                      ..buttonProps = (Button()..style = {'display': 'block'});
```

Comment:
Oh , is this test the only place `Button` is used? If so, we could replace `Button` here with `domProps` or `Dom.div` or something, and remove the copied over Button component.

Sorry, I didn't realize that 😬